server/sonar-db-dao/src/main/java/org/sonar/db/component/BranchDto.java

```diff
@@ -0,0 +1,186 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db.component;
+
+import javax.annotation.Nullable;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class BranchDto {
+  public static final String DEFAULT_MAIN_BRANCH_NAME = "master";
+
+  /**
+   * Maximum length of column "kee"
+   */
+  public static final int KEE_MAX_LENGTH = 255;
+
+  /**
+   * Branch UUID is the projects.uuid that reference projects, branches or pull requests
+   * (projects.qualifier="TRK").
+   * Not null.
+   * Important - the table project_branches does NOT have its own UUIDs for the time being.
+   * All values must exist in projects.uuid.
+   */
+  private String uuid;
+
+  /**
+   * UUID of the project that represents the main branch.
+   * On main branches, projectUuid equals uuid.
+   * Not null.
+   */
+  private String projectUuid;
+
+  /**
+   * Not null.
+   */
+  private BranchKeyType keeType;
+
+  /**
+   * If {@link #keeType} is {@link BranchKeyType#BRANCH}, then name of branch, for example
+   * "feature/foo".
+   *
+   * If {@link #keeType} is {@link BranchKeyType#PR}, then id of the pull request, for
+   * example "1204".
+   */
+  private String kee;
+
+  /**
+   * Value is mandatory when {@link #keeType} is {@link BranchKeyType#BRANCH}.
+   * Otherwise it is null.
+   */
+  @Nullable
+  private BranchType branchType;
+
+  /**
+   * UUID of the branch:
+   * - in which the short-lived branch or pull request will be merged into
+   * - that is the base of long-lived branch.
+   *
+   * Can be null if information is not known.
+   */
+  @Nullable
+  private String mergeBranchUuid;
+
+  /**
+   * Optional title of pull requests
+   */
+  @Nullable
+  private String pullRequestTitle;
+
+  public String getUuid() {
+    return uuid;
+  }
+
+  public BranchDto setUuid(String s) {
+    this.uuid = s;
+    return this;
+  }
+
+  public String getProjectUuid() {
+    return projectUuid;
+  }
+
+  public BranchDto setProjectUuid(String s) {
+    this.projectUuid = s;
+    return this;
+  }
+
+  public boolean isMain() {
+    return projectUuid.equals(uuid);
+  }
+
+  public BranchKeyType getKeeType() {
+    return keeType;
+  }
+
+  public BranchDto setKeeType(BranchKeyType t) {
+    this.keeType = t;
+    return this;
+  }
+
+  /**
+   * This is the getter used by MyBatis mapper.
+   */
+  private String getKee() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "getKee" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)